src/scanners/javascript.js

```diff
@@ -33,13 +33,23 @@ export default class JavaScriptScanner {
           message.ruleId = messages.JS_SYNTAX_ERROR.code;
         }
 
+        var code = message.ruleId.toUpperCase();
+        var messageObj = messages[code];
+
+        // Fallback to looking up the message object by the
+        // constant string passed in context.report()
+        if (!messageObj && message.message) {
```

Comment:
It would be good to add a test for this bit of code. It's implicitly tested by this rule I guess.